pkg/isoeditor/nmstate_handler.go

```diff
@@ -0,0 +1,134 @@
+}
+
+type nmstateHandler struct {
+	workDir  string
+	executer Executer
+}
+
+func NewNmstateHandler(workDir string, executer Executer) NmstateHandler {
+	return &nmstateHandler{
+		workDir:  workDir,
+		executer: executer,
+	}
+}
+
+func (n *nmstateHandler) CreateNmstateRamDisk(rootfsPath, ramDiskPath string) error {
+	// Extract nmstatectl binary
+	var err error
+	nmstateDir := filepath.Join(n.workDir, "nmstate")
+	err = os.MkdirAll(nmstateDir, os.ModePerm)
+	if err != nil {
+		return err
+	}
+	binaryPath, err := n.extractNmstatectl(rootfsPath, nmstateDir)
+	if err != nil {
+		return err
+	}
+	nmstatectlPath := filepath.Join(nmstateDir, "squashfs-root", binaryPath)
+
+	// Check if nmstatectl binary file exists
+	if _, err = os.Stat(nmstatectlPath); os.IsNotExist(err) {
+		return err
+	}
+
+	// Read binary
+	nmstateBinContent, err := os.ReadFile(nmstatectlPath)
+	if err != nil {
+		return err
+	}
+
+	// Create a compressed RAM disk image with the nmstatectl binary
+	compressedCpio, err := generateCompressedCPIO(nmstateBinContent, NmstatectlPathInRamdisk, 0o100_755)
+	if err != nil {
+		return err
+	}
+
+	// Write RAM disk file
+	err = os.WriteFile(ramDiskPath, compressedCpio, 0755) //nolint:gosec
+	if err != nil {
+		return err
+	}
+
+	// Remove temp dir
+	defer func() {
+		removeErr := os.RemoveAll(nmstateDir)
+		if removeErr != nil {
+			log.WithError(removeErr).Error("failed to remove nmstate temp dir")
+		}
+	}()
+
+	return err
+}
+
+func (n *nmstateHandler) extractNmstatectl(rootfsPath, nmstateDir string) (string, error) {
+	_, err := n.executer.Execute(fmt.Sprintf("7z x %s", rootfsPath), nmstateDir)
+	if err != nil {
+		log.Errorf("failed to 7z x rootfs.img: %v", err.Error())
+		return "", err
+	}
+	// limiting files is needed on el<=9 due to https://github.com/plougher/squashfs-tools/issues/125
+	ulimit := "ulimit -n 1024"
+
+	// Listing the filesystem concisely, displaying only files (using `-lc` option).
+	// Each file in the output won't include any prefix before `/ostree` (by using `-dest ''` option),
+	// which is useful when invoking `-extract-file` (after finding the `nmstatectl` binary path).
+	list, err := n.executer.Execute(fmt.Sprintf("%s ; unsquashfs -dest '' -lc %s", ulimit, "root.squashfs"), nmstateDir)
+	if err != nil {
+		log.Errorf("failed to unsquashfs root.squashfs: %v", err.Error())
+		return "", err
+	}
+
+	r, err := regexp.Compile(".*nmstatectl")
+	if err != nil {
+		log.Errorf("failed to compile regexp: %v", err.Error())
+		return "", err
+	}
+	binaryPath := r.FindString(list)
+
+	_, err = n.executer.Execute(fmt.Sprintf("%s ; unsquashfs -no-xattrs %s -extract-file %s", ulimit, "root.squashfs", binaryPath), nmstateDir)
+	if err != nil {
+		log.Errorf("failed to unsquashfs root.squashfs: %v", err.Error())
+		return "", err
+	}
+	return binaryPath, nil
+}
+
+//go:generate mockgen -package=isoeditor -destination=mock_executer.go . Executer
+type Executer interface {
```

Comment:
nit: I believe the correct spelling would be `Executor`

Reply:
Nevermind I might be wrong here, I was just not used to the word "executer" but I think it might be more appropriate